app/services/notifications/new_follower/follow_data.rb

```diff
@@ -1,13 +1,21 @@
 module Notifications
   module NewFollower
-    module Types
-      include Dry.Types
-    end
+    class FollowData
+      class DataError < RuntimeError; end
+
+      include ActiveModel::Model
+      include ActiveModel::Validations
+
+      attr_accessor :followable_id, :followable_type, :follower_id
+
+      validates :followable_id, numericality: { only_integer: true }
+      validates :followable_type, inclusion: { in: %w[User Organization] }
+      validates :follower_id, numericality: { only_integer: true }
 
-    class FollowData < Dry::Struct
-      attribute :followable_id, Types::Strict::Integer
-      attribute :followable_type, Types::Strict::String.enum("User", "Organization")
-      attribute :follower_id, Types::Strict::Integer
+      def initialize(*args, **kwargs)
```

Comment:
The existing code expects `FollowData.new` to raise on invalid data. To reduce the surface area of this change I moved validation into the initializer and raise an exception on invalid data, the same way the `Dry::Struct` code did.